pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,170 @@
+"""Pkgpanda HTTP API"""
+
+import http.client
+import logging
+import sys
+
+from flask import Flask, current_app, jsonify, request
+
+from pkgpanda import Install, Repository, actions
+from pkgpanda.exceptions import PackageError, ValidationError
+
+
+empty_response = ('', http.client.NO_CONTENT)
+
+
+def package_listing_response(package_ids):
+    return jsonify(sorted(package_ids))
+
+
+def package_response(package_id, repository):
+    try:
+        package = repository.load(package_id)
+    except PackageError:
+        error_message = 'Unable to load package {}.'.format(package_id)
+        logging.exception(error_message)
+        return error_response(error_message), http.client.INTERNAL_SERVER_ERROR
+
+    return jsonify({
+        'id': str(package.id),
+        'name': str(package.name),
+        'version': str(package.version),
+    })
+
+
+def error_response(message):
+    return jsonify({'error': message})
+
+
+app = Flask(__name__)
+app.config.from_object('pkgpanda.http.config')
+app.config.from_envvar('PKGPANDA_HTTP_CONFIG', silent=True)
+
+
+@app.errorhandler(Exception)
+def uncaught_exception_handler(exc):
```

Comment:
The CLI has handling for ValidationError, PackageError to top level error messages as they're relatively expected during the normal course of operations. Can we make the flask app return them as proper JSON blob error codes?